models/junos/security_policies_global/junos_security_policies_global.yml

```diff
@@ -0,0 +1,211 @@
+---
+module: junos_security_policies_global
+version_added: 2.12
+short_description: Manage global security policy settings on Juniper JUNOS devices
+description: This module provides declarative management of global security policy settings on Juniper JUNOS devices
+author: Pranav Bhatt (pbhatt@redhat.com)
+requirements:
+  - ncclient (>=v0.6.4)
+notes:
+  - This module requires the netconf system service be enabled on the device being managed.
+  - (to be revisited)This module works with connection C(netconf). See L(the Junos OS Platform Options,https://docs.ansible.com/ansible/latest/network/user_guide/platform_junos.html).
+  - (to be revisited)Tested against JunOS v18.4R1
+options:
+  config:
+    description: A dictionary of security policies
+    type: list
+    elements: dict
+    suboptions:
+      default_policy:
+        description:
+          - Configure the default security policy that defines the actions the device takes on a packet that does not match any user-defined policy.
+        choices:
+          - deny-all
+          - permit-all
+      policy_rematch:
+        description:
+          - Enable the device to reevaluate an active session when its associated security policy is modified. The session remains open if it still matches the policy that allowed the session initially.
+        type: dict
+        suboptions:
+          extensive:
+            description:
+              - When a policy is modified or deleted, extensive option checks if any suitable policy permit to keep these sessions alive.
+            type: bool
+      policy_stats:
+        description:
+          - Configure systemwide policies statistics.
+        type: dict
```

Comment:
This parameter is made dict, with single suboption which's also bool, any specific reason

Reply:
I have tried following the structure used by CLI configuration as much as possible since I thought it would give a more uniform experience. If someone wanted to search what a particular option does, I thought having the RM structure the same would make things easier.

Here in [`policy-stats`](https://www.juniper.net/documentation/us/en/software/junos/security-policies/topics/ref/statement/security-edit-policy-rematch.html) and `policy-rematch`, you can just set those parent parameters to true: 
```
vagrant@vsrx> show configuration security policies
policy-rematch;
policy-stats;
```

You can also set them up with their respective sub-options:
```
vagrant@vsrx> show configuration security policies
policy-rematch;
policy-stats {
system-wide enable;
}
```

This is why I chose to implement certain fields the way you've pointed out. Do I need to make any changes?